streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -561,6 +561,12 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
     }
 
+  /**
+   * Immediately ends the stream without consuming an element
+   */
+  def discard: ZStream[R, E, Unit] =
+    self.take(0).drain
```

Comment:
What's the usecase for this combinator? `take(0)` means the stream does not execute any element effects, just scope effects.

Reply:
Removed. This would again only be useful for eagerly signaling that we are done